лаба2.1

sydak6373 opened the pull request:
реализовал основной функционал приложения заметок

sydak6373 committed:
лаба2.1

POLVRITY commented:
В МРе вижу только удаление ТулБара, другие задачи выполнены не были? Те пункты про которые мы говорили:

- [x]  Реализовать механизм сессии и сохранения текущего пользователя.
- [x]  Реализовать возможность выхода из приложения.

sydak6373 committed:
лаба 2.1

Реализовал вход и регистрацию через URLsession
Переписал функциональную часть заметок с использованием CoreData

POLVRITY commented:
Посмотрел код, есть пара косяков/моментов которые нужно решить:

1) Добавить кнопку выхода на экран "уже залогиненного пользователя" с таблицей, поскольку свайп модального окна на симуляторе делать очень больно и зачастую курсор просто попадает на саму табличку.

2) Добавить флоу очистки поля "пароль" после закрытия экрана "залогиненного пользователя". А то мы получается вроде бы вышли из аккаунта, но логин и пароль все равно остаются на месте.

3) Возможно баг, а возможно серьезная недоработка - Зарегистрировался, зашел на экран, вышел с него, пытаюсь зайти под теми же данными еще раз в "логине" - не работает. Пишет что неверные данные. Нужно разобраться, после чего смогу детальнее сказать по коду.

sydak6373 committed:
доработал функционал

1. исправил пункт 3

2. не до конца понял пункт 1 из за него мог не до конца понять и пункт 2, но вроде и их исправил

3. реализовал выход из приложения

4. Сделал так что теперь не нужно делать вход в аккаунт при каждом заходе в приложение

POLVRITY commented:
https://github.com/sydak6373/Notes/assets/60783971/ebeba5e1-fa50-4481-968a-9db271467a3c

1. Кнопку добавил, отлично.
2. Не исправлено. Прикрепил видеозапись самого бага(поле пароль само заполняется паролем когда я его не ввожу, так быть не должно).
3. Тоже окей.

Осталось поправить только 2ой пункт.

sydak6373 committed:
приложение

исправил пункт 2

POLVRITY commented:
Да, все ок, принято. Только еще нужно конфликты пофиксить при вливании в main ветку.

sydak6373 committed:
Merge branch 'main' into main-review
